File: doc/static/2019/pub/02-kroklinjiga_koordinater/html/fig/streamlines-vortex.m
% Make an x,y grid
[X,Y] = ndgrid(linspace(-2.5,2.5,100),linspace(-2.5,2.5,100));
R=sqrt(X.^2+Y.^2);

% Plot streamlines for corresponding vector field
v_x = - Y ./ R.^2; % Note that -Y/R = -sin(phi)
v_y = X ./ R.^2;   % Note that X/R = cos(phi)

% In Matlab we have to provide start points for streamlines.
start1_y=-2.0:0.5:-0.5;
start_y=[start1_y, -start1_y];
start_x=[zeros(size(start_y))];

% Fine tuning might be needed to get a nice set of streamlines.
strm=streamline(X', Y', v_x', v_y', start_x, start_y,[0.01]);
set(strm,'LineWidth',2,'Color','k')

hold on

% Matlab streamlines have no arrows. Combine with quiver.
% Use fewer grid points to avoid too many arrows in the plot.
[Xc,Yc] = ndgrid(linspace(-3,3,10),linspace(-3,3,10));
Rc=sqrt(Xc.^2+Yc.^2);
v_xc = - Yc ./ Rc.^2; % Note that -Y/R = -sin(phi)
v_yc = Xc ./ Rc.^2;   % Note that X/R = cos(phi)
qvr=quiver(Xc', Yc', v_xc', v_yc','Color','k');

xlabel('x')
ylabel('y')
xlim([-2.5,2.5])
ylim([-2.5,2.5])

File: doc/static/2019/pub/02-kroklinjiga_koordinater/html/fig/streamlines-vortex.py
### Start initialization
import sys
import numpy as np
import matplotlib.pyplot as plt
### End initialization
### Start Alt2: Streamlines with streamplot ###
w = 3
Y, X = np.mgrid[-w:w:100j, -w:w:100j]
R = np.sqrt(X**2+Y**2)
U = -Y / R**2 # Note that -Y/R = -sin(phi)
V = X / R**2  # Note that X/R = cos(phi)

fig = plt.figure()
ax = fig.add_subplot(111)

strm = ax.streamplot(X, Y, U, V, linewidth=2)

ax.set_title(r'Streamplot, vortex field ' +\
        r'$\vec{F} = \frac{J}{2\pi\rho}\hat\varphi$')
ax.set_xlabel('$x$')
ax.set_ylabel('$y$')

ax.set_aspect('equal')

plt.savefig('streamlines-vortex.png')
### End Alt2

### Start Alt1: Streamlines as 2d curves ###
def curve(rho0,x):
    """Fieldline for vortex streamlines in Cartesian coordinates.
    Assume z=0 => Fieldline in xy-plane.
    We also avoid the singularity at x=y=0."""
    if max(abs(x)) > rho0:
        print('Fieldline for rho_0=%3.1f is only defined for |x| <= %3.1f' %(rho0,rho0))
        return None
    return np.sqrt(rho0**2 - x**2)

fig = plt.figure()
ax = fig.add_subplot(111)

# Grid of x points
nx = 64

# Streamline constant
for rho0 in np.linspace(1,3,5):

    # Grid of x points
    x = np.linspace(-rho0,rho0, nx)

    yp = curve(rho0,x)
    ym = -curve(rho0,x)

    # Plot the streamlines with an appropriate colormap and arrow style
    ax.plot(x, yp, 'b-', x, ym, 'b-', linewidth=2)
    ax.annotate("", xy=(max(x), 0.1), xytext=(max(x), 0), arrowprops=dict(arrowstyle="->",color='b'))

ax.set_title(r'Streamplot, vortex field $\vec{F} = \frac{J}{2\pi\rho}\hat\varphi$')
ax.set_xlabel('$x$')
ax.set_ylabel('$y$')

ax.set_aspect('equal')
### End Alt1

#plt.show()
plt.savefig('streamlines-vortex-curves.png')
